File: frontend/src/app/models/ConceptSummary.ts
import {DbEntity} from './DbEntity';
import {ConceptStatus} from './ConceptStatus';
import {Concept} from './Concept';

export class ConceptSummary extends DbEntity {
  context: string;
  fullName: string;
  status: string;
  version: string = "0.1";

  constructor(concept?: Concept) {
    super();

    if (concept) {
      this.id = concept.id;
      this.context = concept.context;
      this.fullName = concept.fullName;
      this.version = concept.version;
    }
  }

}

File: common/src/main/java/org/endeavourhealth/im/common/models/ConceptSummary.java
package org.endeavourhealth.im.common.models;

public class ConceptSummary extends DbEntity<ConceptSummary> {
    private String name;
    private String context;
    private ConceptStatus status;
    private Float version;
    private Boolean synonym;

    public String getName() {
        return name;
    }

    public ConceptSummary setName(String name) {
        this.name = name;
        return this;
    }

    public String getContext() {
        return context;
    }

    public ConceptSummary setContext(String context) {
        this.context = context;
        return this;
    }

    public ConceptStatus getStatus() {
        return status;
    }

    public ConceptSummary setStatus(ConceptStatus status) {
        this.status = status;
        return this;
    }

    public Float getVersion() {
        return version;
    }

    public ConceptSummary setVersion(Float version) {
        this.version = version;
        return this;
    }

    public Boolean getSynonym() {
        return synonym;
    }

    public ConceptSummary setSynonym(Boolean synonym) {
        this.synonym = synonym;
        return this;
    }
}
